Tribler/Core/Modules/restapi/metadata_endpoint.py

```diff
@@ -1,255 +1,151 @@
-            "last": sanitized["last"],
-            "sort_by": sanitized["sort_by"],
-            "sort_asc": int(sanitized["sort_asc"]),
-        })
-
-
-class ChannelsCountEndpoint(BaseChannelsEndpoint):
-
-    def render_GET(self, request):
-        args = recursive_unicode(request.args)
-        sanitized = self.sanitize_parameters(args)
-        return self.get_total_count(self.session.lm.mds.ChannelMetadata, sanitized)
-
-class ChannelsPopularEndpoint(BaseChannelsEndpoint):
-
-    def render_GET(self, request):
-        limit_channels = 10
+        return MetadataPublicKeyEndpoint(self.session, path)
 
-        args = recursive_unicode(request.args)
-        if 'limit' in args and args['limit']:
-            limit_channels = int(args['limit'][0])
-
-            if limit_channels <= 0:
+    def render_PATCH(self, request):
+        try:
+            request_parsed = recursive_unicode(json.twisted_loads(request.content.read()))
+        except ValueError:
+            request.setResponseCode(http.BAD_REQUEST)
+            return json.twisted_dumps({"error": "Bad JSON"})
+        results_list = []
+        for entry in request_parsed:
+            public_key = database_blob(unhexlify(entry.pop("public_key")))
+            id_ = entry.pop("id")
+            error, result = self.update_entry(public_key, id_, entry)
+            # TODO: handle the results for a list that contains some errors in a smarter way
+            if error:
+                request.setResponseCode(error)
+                return json.twisted_dumps(result)
+            results_list.append(result)
+        return json.twisted_dumps(results_list)
+
+    @db_session
+    def render_DELETE(self, request):
+        request_parsed = recursive_unicode(json.twisted_loads(request.content.read()))
+        results_list = []
+        for entry in request_parsed:
+            public_key = database_blob(unhexlify(entry.pop("public_key")))
+            id_ = entry.pop("id")
+            entry = self.session.lm.mds.ChannelNode.get(public_key=public_key, id_=id_)
+            if not entry:
                 request.setResponseCode(http.BAD_REQUEST)
-                return json.twisted_dumps({"error": "the limit parameter must be a positive number"})
-
-        with db_session:
-            popular_channels = self.session.lm.mds.ChannelMetadata.get_random_channels(limit=limit_channels)
-            results = [channel.to_simple_dict() for channel in popular_channels]
-        return json.twisted_dumps({"channels": results})
-
+                return json.twisted_dumps({"error": "Entry %i not found" % id_})
+            entry.delete()
+            result = {"public_key": hexlify(public_key), "id": id_, "state": "Deleted"}
+            results_list.append(result)
+        return json.twisted_dumps(results_list)
 
-class ChannelPublicKeyEndpoint(BaseChannelsEndpoint):
 
+# /<public_key>
+#              /<id_>
+# Intermediate endpoint for parsing public_key part of the request
```

Comment:
Why not place these comments in a docstring under the class definition?

Reply:
Ok